homeassistant/components/androidtv/media_player.py

```diff
@@ -287,6 +292,33 @@ def service_adb_command(service):
         schema=SERVICE_ADB_COMMAND_SCHEMA,
     )
 
+    def service_learn_sendevent(service):
+        """Translate a key press on a remote to ADB 'sendevent' commands."""
+        entity_id = service.data[ATTR_ENTITY_ID]
+        target_devices = [
+            dev
+            for dev in hass.data[ANDROIDTV_DOMAIN].values()
+            if dev.entity_id in entity_id
+        ]
+
+        for target_device in target_devices:
+            output = target_device.learn_sendevent()
+
+            # log the output, if there is any
+            if output:
+                _LOGGER.info(
+                    "Output of 'learn_sendevent' service from '%s': %s",
+                    target_device.entity_id,
+                    output,
+                )
+
+    hass.services.register(
```

Comment:
Don't use `hass.services.register`. Instead use `platform.register_entity_service`.

https://github.com/home-assistant/core/blob/dev/homeassistant/components/sonos/media_player.py#L256-L264

Reply:
I tried to do that, following the example that you linked. However, `entity_platform.current_platform.get()` returns `None`. Maybe it’s because in the Sonos integration it's called in `async_setup_entry`, whereas here it's in `setup_platform`?

Reply:
Ah yes, that is it. You will need to use async_setup_platform to do that. Everything that does I/O then needs to run with `await hass.async_add_executor_job(func_that_does_io, arg1, arg2)`